kubectl/subcommand.go

```diff
@@ -134,6 +134,12 @@ func InspectSubcommand(command string) (Subcommand, bool) {
 		return KubectlPlugin, true
 	}
 
+	command = strings.ReplaceAll(command, "-", "_")
+
+	if _, err := exec.LookPath("kubectl-" + command); err == nil {
+		return KubectlPlugin, true
+	}
```

Comment:
This is very interesting, and made me question if kubectl really acts like this. And you're right, it does.

After looking at their source code and doing some testing, it seems to handle it like this:

- `kubectl do some thing`
  - try exec `kubectl-do-some-thing`
  - try exec `kubectl-do-some thing`
  - try exec `kubectl-do some thing`
- `kubectl do-some thing`
  - try exec `kubectl-do_some-thing`
  - try exec `kubectl-do_some thing`
- `kubectl do-some-thing`
  - try exec `kubectl-do_some_thing`

(Source: <https://github.com/kubernetes/kubectl/blob/d0e936c703558421df2ef367a7c8c1c5972e5c3c/pkg/cmd/cmd.go#L260-L307>)

That's super curious! I never knew!

So with that, then this code in kubecolor should also reflect these use cases.

I suggest you simpky copy-paste the kubectl `HandlePluginCommand` function, but make it return a result if it found the plugin instead of actually executing it. Probably rename it too to something like `IsPluginCommand`

And due to licensing, I suggest placing the `IsPluginCommand` in a separate file, and copy over the apache license header as well as adding a comment that you've modified the code. Adding the URL to the source file would also be preferred for good sport.

Then it'd be great with some tests showing that it works as expected.